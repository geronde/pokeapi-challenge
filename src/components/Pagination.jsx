import React from "react";
import { isNull } from "lodash";
import styled from "styled-components";

const PaginateWrapper = styled.article`
    width: 100%;
    padding: 10px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
  .paginate {
    cursor: pointer;
  }
  .disabled {
    cursor: not-allowed;
  }
  .pagination-counter {
    font-size: 18px;
  }
`;

const Pagination = (props) => {
const { next, previous, totalItems, currentPage, pageSize  } = props.details
  return (
    <PaginateWrapper>
      <div
        onClick={props.previousPage}
        className={!isNull(previous) ? "paginate" : "disabled"}
      >
        <svg
          height="64"
          viewBox="0 0 24 24"
          width="64"
          fill={!isNull(previous) ? "#000" : "#d3d3d3"}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z" />
          <path d="M0-.5h24v24H0z" fill="none" />
        </svg>
      </div>

      <div className="pagination-counter">
        {currentPage} of {Math.ceil(totalItems/pageSize)}
      </div>
      <div
        onClick={props.nextPage}
        className={!isNull(next) ? "paginate" : "disabled"}
      >
        <svg
          height="64"
          fill={!isNull(next) ? "#000" : "#d3d3d3"}
          viewBox="0 0 24 24"
          width="64"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z" />
          <path d="M0-.25h24v24H0z" fill="none" />
        </svg>
      </div>
    </PaginateWrapper>
  );
};

export default Pagination;
