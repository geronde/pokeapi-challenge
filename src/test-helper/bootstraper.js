import faker from 'faker';

export const generateTestPokemons = (n = 15, isType) => {
  const pokemons = [];

  for (let i = 0; i < n; i++) {
    let poke = { name: faker.random.word(), url: `fake-url/${i}/` };
    if (isType) {
      poke = { pokemon: { ...poke } };
    }
    pokemons.push(poke);
  }
  return pokemons;
};
