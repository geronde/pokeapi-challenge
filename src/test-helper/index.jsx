import { render as rtlRender } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

export const render = (ui, { ...renderOptions } = {}) =>{
    function Wrapper({ children }) {
      return <BrowserRouter>{children}</BrowserRouter>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
  }