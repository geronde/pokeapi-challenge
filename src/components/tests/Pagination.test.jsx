import React from "react";
import {render} from '../../test-helper/index'
import Pagination from '../Pagination'

describe('Pagination', () => {
    let props;
    beforeEach(() => {
        props = {
            details: {
                next: '/url/',

                previous: null,
                totalItems: 250,
                currentPage: 1,
                pageSize: 20,
                isFilter: false,
                startIndex: 0,
                endIndex: 19
            },
            previousPage: jest.fn(),
            nextPage: jest.fn()
        }
    });
    afterEach(() => {
        jest.resetAllMocks();
        jest.restoreAllMocks();
    });
    it('should display pagination correctly', () => {
       const { asFragment } = render(<Pagination {...props} />)
       expect(asFragment()).toMatchSnapshot();
    });
    it('should paginate for user', () => {
        props.details.previous = '/url'
        const { container } = render(<Pagination {...props} />)
        expect(container.querySelector('.paginate')).toBeInTheDocument()        
    });
    it('should display next icon as disabled if pagination reach end', () => {
        props.details.next = null;
        const { container } = render(<Pagination {...props} />)
        expect(container.querySelector('.disabled')).toBeInTheDocument()
    });
    it('should display next icon as disabled if pagination at start point', () => {
        props.details.previous = null;
        const { container } = render(<Pagination {...props} />)
        expect(container.querySelector('.disabled')).toBeInTheDocument()
    });
    it('should display next icon as disabled if pagination reach end for isFilter true', () => {
        props.details.isFilter = true;
        props.details.endIndex = 250;
        const { container } = render(<Pagination {...props} />)
        expect(container.querySelector('.disabled')).toBeInTheDocument()
    });
    it('should display next icon as disabled if pagination at start point for isFilter true', () => {
        props.details.isFilter = true;
        props.details.startIndex = 0;
        const { container } = render(<Pagination {...props} />)
        expect(container.querySelector('.disabled')).toBeInTheDocument()
    });
});