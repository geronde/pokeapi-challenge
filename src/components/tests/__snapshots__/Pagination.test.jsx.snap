// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should display pagination correctly 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  padding: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .paginate {
  cursor: pointer;
}

.c0 .disabled {
  cursor: not-allowed;
}

.c0 .pagination-counter {
  font-size: 18px;
}

<article
    class="c0"
  >
    <div
      class="disabled"
      role="button"
    >
      <svg
        fill="#d3d3d3"
        height="64"
        viewBox="0 0 24 24"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
        />
        <path
          d="M0-.5h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
    <div
      class="pagination-counter"
    >
      1 of 13
    </div>
    <div
      class="paginate"
      role="button"
    >
      <svg
        fill="#000"
        height="64"
        viewBox="0 0 24 24"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
        />
        <path
          d="M0-.25h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
  </article>
</DocumentFragment>
`;
